File: java/src/main/java/org/mydotey/scf/type/string/StringToShortConverter.java
package org.mydotey.scf.type.string;

/**
 * @author koqizhao
 *
 * May 21, 2018
 */
public class StringToShortConverter extends StringConverter<Short> {

    public static final StringToShortConverter DEFAULT = new StringToShortConverter();

    public StringToShortConverter() {
        super(Short.class);
    }

    @Override
    public Short convert(String source) {
        return Short.parseShort(source);
    }

}

File: dotnet/src/MyDotey.SCF.Simple/Type/String/StringToShortConverter.cs
using System;

namespace MyDotey.SCF.Type.String
{
    /**
    * @author koqizhao
    *
    * May 21, 2018
    */
    public class StringToShortConverter : StringConverter<short?>
    {
        public static readonly StringToShortConverter Default = new StringToShortConverter();

        public override short? Convert(string source)
        {
            if (string.IsNullOrWhiteSpace(source))
                return null;

            return short.Parse(source);
        }
    }
}